stdlib/Distributed/src/remotecall.jl

```diff
@@ -351,7 +359,10 @@ end
 
 channel_type(rr::RemoteChannel{T}) where {T} = T
 
-serialize(s::ClusterSerializer, f::Future) = serialize(s, f, f.v === nothing)
+function serialize(s::ClusterSerializer, f::Future)
+    v_cache = @atomic f.v
```

Comment:
It seems a bit aggressive to make serialize/deserialize full atomic barriers (instead of their defaults). @tkf any thoughts too?

Reply:
fyi this is just one leftover `future.v` usage that wasn't wrapped in `@atomic` and didn't error before in tests or usage

Reply:
Hmm... Can `serialize(s::ClusterSerializer, f::Future)` call `serialize(::AbstractSerializer, ::typeof(something(f.v)))`? If so, I guess we need to make it as strong as `fetch` since the author of `MyType` can override `serialize(::AbstractSerializer, ::MyType)` that can contain arbitrary code?

Reply:
Looking closer, the issue here is that we need a lock around `f.v` for the duration of this function

Reply:
put a lock on `f.lock` around the full serialize function